### PR TITLE
 Command line deployment to Azure and deployment scenario visibility

### PR DESCRIPTION
**Pull request by CamSoper**
Addresses https://github.com/dotnet/docs/issues/2716

[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-azure-webapp-using-cli?branch=pr-en-us-4728)

### CODE REVIEW

**Comment by CamSoper** [comment]
Hey @Rick-Anderson @rachelappel @scottaddie , which GitHub team do I need to be on to not get the cla-required warning, and change labels, etc.?

Also, does this repo not automatically stage the docs anywhere for review?  I'm not done with the PR yet, but I wanted to get this article on a staging server before I started making changes to the TOC or add the deployment links to other tutorials that Mads requested.

**Comment by Rick-Anderson** [comment]
@Eilon  can you add @CamSoper  to asp.net so he doesn't get the CLA -error?

**Comment by Rick-Anderson** [comment]
@CamSoper  I think we need to get this is the TOC too.

File: aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+* [Git](https://www.git-scm.com/) command line client
+
+## Create a web application
+
+Create a new directory for the web application, create a new ASP.NET Core MVC application, and then run the website locally.
+
+```cmd
+REM Create the directory and navigate to it.
+REM These commands are Windows-specific. The Bash equivalents are mkdir and cd.
+md MyApplication
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md]: prefer `dotnet new razor -o aspnetcoreapp`
use Razor, not MVC and create the output dir with the command

File: aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+REM Create a new ASP.NET Core MVC application
+dotnet new mvc
+
+REM Run the application
+dotnet run
+```
+
+![Command line output](publish-to-azure-webapp-using-cli/_static/new_prj.png)
+
+You can then test the application by browsing to the displayed address (usually `http://localhost:5000`).
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md]: I've pretty much banned "you" and it's forms from this repo. How about
Test the app by browsing to http://localhost:5000
Notes:
I think it's always 5000
Don't code fence it, that way they can click on the link

File: aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+az group create --name DotNetAzureTutorial --location EastUS
+
+# Create an App Service plan.
+az appservice plan create --name $webappname --resource-group DotNetAzureTutorial --sku FREE
+
+# Create the Web App
+az webapp create --name $webappname --resource-group DotNetAzureTutorial --plan $webappname
+```
+
+Before you deploy, you need to set the account-level deployment credentials.  Use the script below, making sure to include your own values for the user name and password.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md]: Before deployment, set the account-level deployment credentials using the following script:

Or,

Before deployment, set the account-level deployment credentials using the following script. Replace the "<desired user name>" and ... 

But I don't think we need to tell them.

**Comment by CamSoper** [comment]
@Rick-Anderson I agree.  I just hadn't gotten that far yet.  I've got the change to the TOC and some visibility links in other tutorials (e.g., Next Steps:  Deploy your application using VS or the CLI) I'm still working on.  :)

File: aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+```azurecli-interactive
+az webapp deployment user set --user-name <desired user name> --password <desired password>
+```
+
+You'll also need the deployment URL to deploy the application using Git.  Retrieve the URL like this.
+
+```azurecli-interactive
+az webapp deployment source config-local-git -n $webappname -g DotNetAzureTutorial --query [url] -o tsv
+```
+Note the displayed URL ending in `.git`.  You'll need it in the next step.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md]: Total ban on future tense.
You need it in the next step. But better yet, get rid of you.

Note the displayed URL ending in `.git`; it's used it in the next step.

File: aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+## Deploy the application using Git
+
+You're ready to deploy from your local machine using Git.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md]: The app is ready to deploy ...

File: aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+git commit -a -m "Initial commit"
+
+REM Add the URL as a Git remote repository
+git remote add azure <THE GIT URL YOU NOTED EARLIER>
+
+REM Push the local repository to the remote
+git push azure master
+```
+
+You will be prompted for the deployment credentials you set earlier.  After authenticating, the application will be built remotely and deployed.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/publish-to-azure-webapp-using-cli.md]: no future tense

**Comment by Eilon** [comment]
@Rick-Anderson / @CamSoper - there's no magic list, you have to sign the CLA.

**Comment by Eilon** [comment]
There ya go 😄

**Comment by CamSoper** [comment]
@Rick-Anderson Requested changes made, added to TOC, added some text and a few links to increase visibility.  We've got publishing to Azure mentioned now at the end of every tutorial except the EF tutorials (didn't seem like a great fit), but the EF tutorials do link to the publishing page.  The "Advanced Topics" EF page might benefit from a bit about publishing to Azure, but it's going to be a tiny bit complicated by the addition of SQL Database.

**Comment by CamSoper** [comment]
@Rick-Anderson Is there something special about how the TOC works in your repo?  The new TOC entry is showing up, but it's not clickable, nor is it highlighted when viewing the new article...

**Comment by Rick-Anderson** [comment]
@CamSoper  looks great!

**Pull request merged**